packages/eslint-plugin-turbo/lib/configs/recommended.ts

```diff
@@ -1,17 +1,12 @@
 import { RULES } from "../constants";
-import getEnvVarDependencies from "../utils/getEnvVarDependencies";
+import { Project } from "../utils/calculate-inputs";
+
+const project = new Project(process.cwd());
+const cacheKey = project.valid() ? project.key() : Math.random();
```

Comment:
This `Math.random()` call makes sure that you don't get cache collisions if we fail to detect your project. That means you can keep trying until your project is correct.

(this fixes an edge case)